gensim/models/word2vec.py

```diff
@@ -562,14 +573,6 @@ def scale_vocab(self, min_count=None, sample=None, dry_run=False):
         # print extra memory estimates
         report_values['memory'] = self.estimate_memory(vocab_size=len(retain_words))
 
-        if not dry_run:
-            new_vocab = {}
-            for w in retain_words:
-                new_vocab[w] = self.vocab[w]
-                new_vocab[w].index = len(self.index2word)
-                self.index2word.append(w)
-            self.vocab = new_vocab
-            logger.info("vocabulary min_count & sample applied, and indexes assigned")
         return report_values
```

Comment:
Alright! I've postponed the release for until we settle this. I will increase the max_vocab_size to 2m and "forget" `raw_vocab` after `scale_vocab`.

And then finally release :)

We can add more options / functionality later of course.